VMD_scripts/blobulation.tcl

```diff
@@ -0,0 +1,436 @@
+
+
+proc blobulate {MolID lMin H} {
+	#
+	#	The overarching proc, users use this to run program
+	#
+	#
+	#	Arguments:
+	#	MolID (Integer): An integer that assigns what protein the algorithm looks for 
+	#	lMin (Integer): An integers greater than 1 and less then the legnth of the sequence that determines the minimum length of hblobs
+	# 	H (Float): A float that determines the hydropathy threshold, this determines how hydrophobic something needs to be to be counted
+	#	for an h blob
+	#
+	#	Results:
+	#	The results is a user value applied to the protein of choice the differentiates h blobs, p blobs, and s blobs. 
+	
+	source normalized_hydropathyscales.tcl
+	set checked [checker $MolID $lMin $H]
+	if {$checked == -1} {
+		puts "Variables are incorrect ending program"
+		return  
+		}
+	if {$checked == 1} { 
+		puts "here!"
+		set lower [string tolower $MolID]
+		set sel [atomselect $lower alpha]
+		set sorted [lsort -unique [$sel get chain]]
+		puts $sorted
+
+		set chainBlobs {}
+		puts "here 2!"
+		for {set i 0} {$i <= [expr [llength $sorted] -1 ] } { incr i} {
+			puts $chainBlobs
+			set Chain [lindex $sorted $i] 
+			set blobulated [blobulateChain $MolID $lMin $H $Chain]
+			foreach bb $blobulated {
+				lappend chainBlobs $bb
+				
+			} 
+		}
+		if {$chainBlobs != -1} {
+		set lower [string tolower $MolID]
+		set sel [atomselect $lower alpha]
+		$sel set user $chainBlobs
+		$sel get user
+		$sel delete
+		} 
+		return $chainBlobs
+		}
+	
+	set sequence [getSequence $MolID]
+	set hydroS [hydropathyScores $KD_Normalized $sequence]
+	if {$hydroS == -1} {
+		return -1
+		}
+	set hydroM [hydropathyMean $hydroS $sequence]
+	set dig [Digitize $H $hydroM ]
+	set blobh [ blobH $dig $lMin ]
+	set blobs [ blobS $blobh $dig $lMin ]
+	set blobp [ blobP $blobs $dig ]
+    	set blobulated [blobAssign $blobp]
```

Comment:
This looks like blobulated consists solely of p-blobs. Is that the case here?

Reply:
I have changed all proc name to what the proc creates, so blobS is now hsBlob, since it's making a list with both in it. blobP is now named hpsBlob.